src/components/SlateEditor/plugins/uuDisclaimer/SlateDisclaimer.tsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { ReactNode, useCallback, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Editor, Element, Path, Transforms } from "slate";
+import { ReactEditor, RenderElementProps } from "slate-react";
+import styled from "@emotion/styled";
+import { IconButtonV2 } from "@ndla/button";
+import { colors, spacing } from "@ndla/core";
+import { Pencil } from "@ndla/icons/action";
+import { Modal, ModalCloseButton, ModalContent, ModalHeader, ModalTitle, ModalTrigger } from "@ndla/modal";
+import { UuDisclaimerEmbedData, UuDisclaimerMetaData } from "@ndla/types-embed";
+import { UuDisclaimerEmbed } from "@ndla/ui";
+import DisclaimerForm from "./DisclaimerForm";
+import { DisclaimerElement, TYPE_DISCLAIMER } from "./types";
+import DeleteButton from "../../../DeleteButton";
+
+interface Props {
+  attributes: RenderElementProps["attributes"];
+  children: ReactNode;
+  editor: Editor;
+  element: DisclaimerElement;
+}
+
+const DisclaimerBlockContent = styled.div`
+  border: 1px solid ${colors.brand.primary};
+  margin-top: ${spacing.xsmall};
+  padding: ${spacing.xsmall};
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  position: absolute;
+  right: -${spacing.large};
+`;
+
+const StyledModalHeader = styled(ModalHeader)`
+  padding-bottom: 0px;
+`;
+
+const SlateDisclaimer = ({ attributes, children, element, editor }: Props) => {
+  const { t } = useTranslation();
+  const { data } = element;
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+
+  const embed: UuDisclaimerMetaData | undefined = useMemo(
+    () =>
+      data
+        ? {
+            status: "success",
+            data: { disclaimerLink: { text: "Disclaimer link", href: "https://ndla.no" } },
+            embedData: { ...data, disclaimer: data?.disclaimer },
+            resource: data?.resource,
+          }
+        : undefined,
```

Comment:
Skjønner at dette er for test, men tenker det er bedre å ha en default-tekst i defaultDisclaimerBlock henta fra phrases. Feks "UU-advarsel. Bytt ut teksten med din egen beskrivelse.", eller lignande.